From 3d160d1a29483095c77457429c6514f682f145c4 Mon Sep 17 00:00:00 2001
From: Xiaolin He <xiaolin.he@nxp.com>
Date: Mon, 1 Apr 2019 13:12:42 +0800
Subject: [PATCH] Modify CFLAGS, LDFAGS and SRCS macros for tsn

Signed-off-by: Xiaolin He <xiaolin.he@nxp.com>
---
 Makefile.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 1ea9815..f343003 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -17,9 +17,9 @@ INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA = @INSTALL_DATA@
 LIBS = @LIBS@
-CFLAGS = -Wall -pthread @CFLAGS@
+CFLAGS = -Wall -pthread @CFLAGS@ $(shell pkg-config --cflags libnl-3.0 libnl-genl-3.0 libcjson)
 CPPFLAGS = @CPPFLAGS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -ltsn $(shell pkg-config --libs libnl-3.0 libcjson) -lpthread -lrt
 LIBTOOL = $(libtool) --tag=CC --quiet
 
 NETOPEER_MANAGER = @NETOPEER_MANAGER@
@@ -31,7 +31,7 @@ MODEL = @PROJECTNAME@.yin \
 	@PROJECTNAME@-schematron.xsl
 
 
-SRCS = @PROJECTNAME@.c
+SRCS = @PROJECTNAME@.c parse_qbv_node.c xml_node_access.c
 OBJDIR = .obj
 LOBJS = $(SRCS:%.c=$(OBJDIR)/%.lo)
 
-- 
2.17.1

