From e2bf0775746346e90ca206c25b46d69cd4842643 Mon Sep 17 00:00:00 2001
From: Liu Po-B43644 <Po.Liu@freescale.com>
Date: Wed, 22 Mar 2017 17:59:53 +0800
Subject: [PATCH 4/4] Debug: check module config folder is what

Signed-off-by: Liu Po-B43644 <Po.Liu@freescale.com>
---
 server/src/cfgnetopeer_transapi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/server/src/cfgnetopeer_transapi.c b/server/src/cfgnetopeer_transapi.c
index 91afb01..84e1871 100644
--- a/server/src/cfgnetopeer_transapi.c
+++ b/server/src/cfgnetopeer_transapi.c
@@ -226,6 +226,7 @@ int module_enable(struct np_module* module, int add) {
 		nc_verb_error("asprintf() failed (%s:%d).", __FILE__, __LINE__);
 		return(EXIT_FAILURE);
 	}
+	printf("module_enable: config_path is %s\n", config_path);
 	if ((module_config = xmlReadFile(config_path, NULL, XML_PARSE_NOBLANKS|XML_PARSE_NSCLEAN|XML_PARSE_NOWARNING|XML_PARSE_NOERROR)) == NULL) {
 		nc_verb_error("Reading configuration for %s module failed", module->name);
 		free(config_path);
-- 
1.8.3.1

