From 85f52cdcf54078dc5c4a663ff482e1d4d337b372 Mon Sep 17 00:00:00 2001
From: Liu Po-B43644 <Po.Liu@freescale.com>
Date: Wed, 22 Mar 2017 02:20:10 +0800
Subject: [PATCH 1/1] yang-sja1105: modify configure file pass buildroot

Signed-off-by: Liu Po-B43644 <Po.Liu@freescale.com>
---
 Makefile.in  | 20 ++++++++++++++------
 configure.in |  6 +++---
 2 files changed, 17 insertions(+), 9 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index bfc0f07..ec29c70 100755
--- a/Makefile.in
+++ b/Makefile.in
@@ -49,18 +49,26 @@ $(OBJDIR)/%.lo: %.c
 install: $(TARGET)
 	$(INSTALL) -m 775 -d $(DESTDIR)/$(libdir)
 	$(LIBTOOL) --mode=install cp $(TARGET) $(DESTDIR)/$(libdir)/;
-	$(INSTALL) -d $(NETOPEER_DIR)/$(PKGNAME)/
+	$(INSTALL) -d $(DESTDIR)/$(NETOPEER_DIR)/$(PKGNAME)/
 	@for i in $(MODEL); do \
-		$(INSTALL_DATA) -m 600 $$i $(NETOPEER_DIR)/$(PKGNAME)/; \
+		$(INSTALL_DATA) -m 600 $$i $(DESTDIR)/$(NETOPEER_DIR)/$(PKGNAME)/; \
 	done
 	if test -n "$(NETOPEER_MANAGER)"; then \
-		if test -n "`$(NETOPEER_MANAGER) list | grep "^$(PKGNAME) ("`"; then \
-			$(NETOPEER_MANAGER) rm --name $(PKGNAME); \
+		if test -n "`$(NETOPEER_MANAGER) list \
+				--modules_path $(DESTDIR)/$(NETOPEER_DIR)/modules.conf.d/ \
+				--netopeer_config $(DESTDIR)/$(NETOPEER_DIR)/cfgnetopeer/datastore.xml | grep "^$(PKGNAME) ("`"; then \
+			$(NETOPEER_MANAGER) rm --name $(PKGNAME) \
+			--modules_path $(DESTDIR)/$(NETOPEER_DIR)/modules.conf.d/ \
+			--netopeer_config $(DESTDIR)/$(NETOPEER_DIR)/cfgnetopeer/datastore.xml;\
 		fi; \
 		$(NETOPEER_MANAGER) add --name $(PKGNAME) \
-			--model $(NETOPEER_DIR)/$(PKGNAME)/$(PKGNAME).yin \
+			--model $(DESTDIR)/$(NETOPEER_DIR)/$(PKGNAME)/$(PKGNAME).yin \
+			--model_target $(NETOPEER_DIR)/$(PKGNAME)/$(PKGNAME).yin \
 			--transapi $(DESTDIR)/$(libdir)/$(PKGNAME).so \
-			--datastore $(NETOPEER_DIR)/$(PKGNAME)/datastore.xml; \
+			--datastore $(NETOPEER_DIR)/$(PKGNAME)/datastore.xml \
+			--transapi_target $(libdir)/$(PKGNAME).so \
+			--modules_path $(DESTDIR)/$(NETOPEER_DIR)/modules.conf.d/ \
+			--netopeer_config $(DESTDIR)/$(NETOPEER_DIR)/cfgnetopeer/datastore.xml;\
 	fi
 
 .PHONY: uninstall
diff --git a/configure.in b/configure.in
index 05d6efd..ca745e4 100644
--- a/configure.in
+++ b/configure.in
@@ -65,7 +65,7 @@ if test -z "$WITH_LIBXML2" ; then
 	if test "$XML2_CONFIG" = "no"; then
 		AC_MSG_ERROR([Missing development package of libxml2.])
 	fi
-	AC_CHECK_LIB([xml2], [main], [LIBS="`xml2-config --libs` $LIBS" CFLAGS="`xml2-config --cflags` $CFLAGS"], AC_MSG_ERROR([Libxml2 not found ]))
+	AC_CHECK_LIB([xml2], [main], [LIBS="`${WITH_LIBXML2}/xml2-config --libs` $LIBS" CFLAGS="`${WITH_LIBXML2}/xml2-config --cflags` $CFLAGS"], AC_MSG_ERROR([Libxml2 not found ]))
 fi
 
 # Checks for header files.
-- 
1.8.3.1

